src/app_view.js

```diff
@@ -0,0 +1,277 @@
+/**
+ * @license
+ * Blockly Demos: Block Factory
+ *
+ * Copyright 2017 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+goog.require('BlockFactory');
+goog.require('FactoryUtils');
+goog.require('BlockLibraryController');
+goog.require('BlockExporterController');
+goog.require('goog.dom.classlist');
+goog.require('goog.ui.PopupColorPicker');
+goog.require('goog.ui.ColorPicker');
+
+/**
+ * @fileoverview The AppView Class deals with the visual parts of the main
+ * devtools application, such as the menubar.
+ *
+ * @author celinechoo (Celine Choo)
+ */
+class AppView {
+  constructor() {
+    // Main window of application.
+    this.win = nw.Window.get();
+
+    // Initializes menu structure. Leaf nodes are actionable MenuItems.
+    this.menuTree = {
+      'File': {
+        'New': {
+          'New Project': null,
+          'New Block': null,
+          'New Library': null,
+          'New Toolbox': null
+        },
+        'Open Project': null,
+        'Save': null,
+        'Save Only': {
```

Comment:
Fine for now, but as I look at these two, I'm wondering if these will be dependent on the project configuration. One has to configure the locations of each platform before one can save it. A project may only have iOS configured, so does it make sense to have a save item for the others? Maybe the others are just disabled, or maybe they trigger the project configuration screen.

Also, It might read cleaner to make "Save" the menu group, and add an "All" at the top.